### PR TITLE
fix(IItem): make id and owner the only required properties of IItem

### PR DESCRIPTION
**Pull request by jgravois**
AFFECTS PACKAGES:
@esri/arcgis-rest-common-types

ISSUES CLOSED: #171

### CODE REVIEW

**Comment by jgravois** [comment]
whoops.

i didn't see any errors locally because i didn't run `npm run build` after making the change.

**Comment by noahmulfinger** [comment]
@jgravois Looks like fixing #171 requires a more fine-grained approach. We should probably just make all `IItem` properties optional and then extend `IItem` whenever a function requires certain properties.

**Comment by tomwayson** [comment]
Good idea @noahmulfinger - I suggest something like`IUpdateItemParams` which inherits from `IItem`

**Comment by jgravois** [comment]
> We should probably just make all IItem properties optional

Agreed.

a perfect example of why is `createItemInFolder()`. up until this morning it ignored `item.owner` entirely in favor of `requestOptions.owner` or `requestOptions.authentication.username`

i've tacked on another commit which includes the value in the decision tree (with more precedence than either of the other two sources).

**Comment by jgravois** [comment]
after chatting w/ @tomwayson briefly i'm convinced that the correct order of precedence is

1. requestOptions.owner
2. requestOptions.item.owner
3. requestOptions.authentication.username

i did a little manual testing and confirmed that regardless of whether `item.owner` is omitted in the request or contains a value that doesn't match the url route, the REST API will overwrite with the appropriate JSON. i'm happy about that.

File: packages/arcgis-rest-items/src/items.ts [hunk 1]
```diff
@@ -34,10 +34,6 @@ export interface IItemJsonDataRequestOptions extends IItemIdRequestOptions {
    * JSON object to store
    */
   data: any;
-  /**
```
Comment (jgravois) [inline packages/arcgis-rest-items/src/items.ts]: removed a duplicate property present in the interface being extended.

**Pull request merged**